src/components/ButtonMenu.tsx

```diff
@@ -9,31 +10,42 @@ import {
   OverlayTrigger,
   OverlayTriggerProps,
 } from "src/components/internal/OverlayTrigger";
+import { Css } from "src/Css";
+import { TextField } from "src/inputs";
 import { useTestIds } from "src/utils";
 
 interface ButtonMenuProps extends Pick<OverlayTriggerProps, "trigger" | "placement" | "disabled" | "tooltip"> {
   items: MenuItem[];
   persistentItems?: MenuItem[];
+  searchable? : boolean;
   // for storybook purposes
   defaultOpen?: boolean;
 }
 
 export function ButtonMenu(props: ButtonMenuProps) {
-  const { defaultOpen, disabled, items, persistentItems, trigger } = props;
+  const { defaultOpen, disabled, items, persistentItems, trigger, searchable } = props;
   const state = useMenuTriggerState({ isOpen: defaultOpen });
+  const [searchTerm, setSearchTerm] = useState<string | undefined>("");
   const buttonRef = useRef<HTMLButtonElement>(null);
   const { menuTriggerProps, menuProps } = useMenuTrigger({ isDisabled: !!disabled }, state, buttonRef);
   const tid = useTestIds(
     props,
     isTextButton(trigger) ? trigger.label : isIconButton(trigger) ? trigger.icon : trigger.name,
   );
+  
+  // filter our items by our search term
+  const filteredItems = searchTerm ? match(searchTerm, items) : items;
+
 
   return (
     <OverlayTrigger {...props} menuTriggerProps={menuTriggerProps} state={state} buttonRef={buttonRef} {...tid}>
+      { searchable && (
+          <TextField placeholder="Search" value={searchTerm} onChange={setSearchTerm} label="" />
+      )}
       <Menu
         ariaMenuProps={menuProps}
         onClose={() => state.close()}
-        items={items}
+        items={filteredItems}
         persistentItems={persistentItems}
         {...tid}
       />
```

Comment:
**thought:** I'm thinking it might be better off having this combination of a Menu + Search be its own component. Fewer things to rerender when the state updates. Right now we re-render the whole ButtonMenu component whenever the `searchTerm` updates, as this components is in charge of that state. I'd be curious to see if it is worth adding `searchable` to the Menu component itself, or if we should make a `SearchableMenu` component that is basically these two things. I'd say take a look and see what makes more sense.

Reply:
Just took a look at Storybook, might be better doing this directly in the Menu component so we can get the search box to appear per designed within the menu.
Also, I am thinking we might not want to use the `TextField` component for this as the design of the input is very different than what we get from `TextField`. Maybe another text field component that is bespoke for this purpose, "MenuSearchField" perhaps?